src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.DegreesToRadians.cs

```diff
@@ -25,10 +26,59 @@ public static void DegreesToRadians<T>(ReadOnlySpan<T> x, Span<T> destination)
         private readonly struct DegreesToRadiansOperator<T> : IUnaryOperator<T, T> where T : ITrigonometricFunctions<T>
         {
             public static bool Vectorizable => true;
+
             public static T Invoke(T x) => T.DegreesToRadians(x);
-            public static Vector128<T> Invoke(Vector128<T> x) => (x * T.Pi) / T.CreateChecked(180);
-            public static Vector256<T> Invoke(Vector256<T> x) => (x * T.Pi) / T.CreateChecked(180);
-            public static Vector512<T> Invoke(Vector512<T> x) => (x * T.Pi) / T.CreateChecked(180);
+
+            public static Vector128<T> Invoke(Vector128<T> x)
+            {
+#if NET9_0_OR_GREATER
+                if (typeof(T) == typeof(double))
+                {
+                    return Vector128.DegreesToRadians(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Vector128.DegreesToRadians(x.AsSingle()).As<float, T>();
+                }
+#else
+                return (x * T.Pi) / T.CreateChecked(180);
```

Comment:
For many of these the code required for net9.0 is much less than for net8.0, but here the net8.0 code is so concise. Is it worth the extra complexity to ifdef this?

Reply:
We get additional benefits with regards to an inlining profitability boost and the general ability to guarantee that they stay consistent if the algorithm changes or is optimized for future platforms.